python/dgl/distributed/graph_partition_book.py

```diff
@@ -1176,26 +1206,28 @@ class PartitionPolicy(object):
     Parameters
     ----------
     policy_str : str
-        Partition policy name, e.g., 'edge:_E' or 'node:_N'.
+        Partition policy name, e.g., 'edge~_N:_E:_N' or 'node~_N'.
     partition_book : GraphPartitionBook
         A graph partition book
     """
 
     def __init__(self, policy_str, partition_book):
-        splits = policy_str.split(":")
+        splits = policy_str.split(POLICY_DELIMITER)
         if len(splits) == 1:
             assert policy_str in (
                 EDGE_PART_POLICY,
                 NODE_PART_POLICY,
             ), "policy_str must contain 'edge' or 'node'."
             if NODE_PART_POLICY == policy_str:
-                policy_str = NODE_PART_POLICY + ":" + DEFAULT_NTYPE
+                policy_str = NODE_PART_POLICY + POLICY_DELIMITER + DEFAULT_NTYPE
             else:
-                policy_str = EDGE_PART_POLICY + ":" + DEFAULT_ETYPE[1]
+                policy_str = EDGE_PART_POLICY + POLICY_DELIMITER + DEFAULT_ETYPE[1]
         self._policy_str = policy_str
         self._part_id = partition_book.partid
         self._partition_book = partition_book
-        self._type_name = _str_to_tuple(self.policy_str[5:])
+        self._type_name = self.policy_str[5:]
```

Comment:
I will suggest merge this line with L1215. E.g.,
```
policy_name, self._type_name = policy_str.split(POLICY_DDELIMITER)
```

You can also use regex to perform more rigorous matching (and checking).